### ISSUE TITLE
My Regression loss always 0

### ISSUE THREAD

**Issue by shekarneo** [zylo117/Yet-Another-EfficientDet-Pytorch#458]
when i try training on custom data from scratch/head only regression loss always 0

**Comment by HaoyangRong** [comment]
Having the same issue here.

**Comment by zylo117** [comment]
Refer to FAQ Q3.
https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/issues/386

**Comment by shekarneo** [comment]
Why category id matters for regression loss.

**Comment by shekarneo** [comment]
i have tried pre trained b4 with 50 images but loss did not came down from 450 for 50 epochs, 
also tried train from scratch for 200 epochs got the same results.

while signatrix model gave pretty much good results..

may be i should try only b0 model?

**Comment by mytk2012** [comment]
> Why category id matters for regression loss.

This is for the COCO data format.

**Comment by HaoyangRong** [comment]
Ok, I figured it out. In the annotation json files, the 'image_id's in field 'annotations' is a mapping to the 'id's in field 'images'. And it seems both 'image_id'(under 'annotations') and 'id'(under 'images') should be ints.
If we can use 'image_id' instead of 'id' in field 'images', it should be less confusing. Or is it possible to add some format checks?

**Comment by shekarneo** [comment]
did you got better accuracy, i am getting loss 300-400 it is not decreasing even for more than 100 epochs

**Comment by a554142589** [comment]
> did you got better accuracy, i am getting loss 300-400 it is not decreasing even for more than 100 epochs

hello, Had you fixed the error about the "zero-reg" loss?
